rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/client/LRAIT.java

```diff
@@ -0,0 +1,192 @@
+        assertNull("remote invocations should not update current (see MP-LRA issue behaviour)", lraClient.getCurrent());
+    }
+
+    /**
+     * Test that remote invocations can start and stop LRAs using the prorprietary client API
+     */
+    @Test
+    public void testRemoteCurrent() {
+        URI lra1 = invokeInTransaction(null, LRAParticipant.RESOURCE_PATH, LRAParticipant.CREATE_OR_CONTINUE_LRA2, 200);
+        invokeInTransaction(lra1, LRAParticipant.RESOURCE_PATH, LRAParticipant.END_EXISTING_LRA, 200);
+    }
+
+    /**
+     * JBTM-3223 Validate LRA behaviour when multiple LRA's are active on the same thread
+     *
+     * when multiple LRA's are active on the same thread that as one LRA finishes that:
+     * a) the next one is handled correctly
+     * b) creating another one still works fine
+     */
+    @Test
+    public void testCurrent() throws URISyntaxException {
+        NarayanaLRAClient lraClient = new NarayanaLRAClient();
+
+        // start two LRAs on the current thread
+        URI lra1 = lraClient.startLRA("lra1");
+        assertEquals("lra1 is not associated with the current thread",
+                lra1, lraClient.getCurrent());
+        URI lra2 = lraClient.startLRA("lra2");
+        assertEquals("lra2 is not associated with the current thread",
+                lra2, lraClient.getCurrent());
+
+        // a) closing the current one should make the previous one active
+        lraClient.closeLRA(lra2);
+        assertEquals("closing the current LRA should have made the previous one active",
+                lra1, lraClient.getCurrent());
+
+        // b) verify that creating another LRA still works fine
+        URI lra3 = lraClient.startLRA("lra3");
+        assertEquals("lra3 is not associated with the current thread",
+                lra3, lraClient.getCurrent());
+
+        lraClient.closeLRA(lra3);
+        assertEquals("closing the current LRA (lra3) should have made the previous one (lra1) active",
+                lra1, lraClient.getCurrent());
+
+        // check that closing the last LRA will leave none active
+        lraClient.closeLRA(lra1);
+        assertNull("all LRAs are closed so none should be associated with the calling thread",
+                lraClient.getCurrent());
+
+        lraClient.close();
+    }
+
+    private URI invokeInTransaction(URI lra, String resourcePrefix, String resourcePath, int expectedStatus) {
+        Response response = null;
+
+        try {
+            Invocation.Builder builder = client.target(UriBuilder.fromUri(baseURL.toExternalForm())
+                    .path(resourcePrefix)
+                    .path(resourcePath).build())
+                    .request();
+
+            if (lra != null) {
+                builder.header(LRA_HTTP_CONTEXT_HEADER, lra.toASCIIString());
+            }
+
+            response = builder.get();
+
+            assertEquals(expectedStatus, response.getStatus());
+            assertTrue(response.hasEntity());
```

Comment:
it could be good to add some more context about the failure to the assertion message